tox.ini

```diff
@@ -0,0 +1,23 @@
+# Tox (http://tox.testrun.org/) is a tool for running tests
+# in multiple virtualenvs. This configuration file will run the
+# test suite on all supported python versions. To use it, "pip install tox"
+# and then run "tox" from this directory.
+
+[tox]
+isolated_build = true
+envlist = clean, py{38,39,310,311,32}-all-extras
+
+[testenv]
+skip_install = true
+allowlist_externals = poetry
+setenv =
+   passenv = RUNNERNAME
+   # setting this value to use system environment path.
+   ProgramW6432 = C:/Program Files
```

Comment:
I don't understand why you're overriding this.

Reply:
Is this error the reason you are setting ProgramW6432 or is it the result of you setting ProgramW6432?

```
E           KeyError: 'ProgramW6432'
C:\Program Files\Python39\lib\os.py:679: KeyError
_ ERROR at setup of test___multi_session___measure___reserves_multiple_sessions[configuration10] _
    @pytest.fixture(scope="session")
    def discovery_service_process() -> Generator[DiscoveryServiceProcess, None, None]:
        """Test fixture that creates discovery service process."""
        if sys.platform != "win32":
            pytest.skip(
                f"Platform {sys.platform} is not supported for starting discovery service."
                "Could not proceed to run the tests."
            )
>       elif not _get_registration_json_file_path().exists():
tests\conftest.py:68: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
ni_measurementlink_service\discovery\_support.py:59: in _get_registration_json_file_path
    pathlib.Path(os.environ["ProgramW6432"])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = environ({'COMSPEC': 'C:\\Windows\\system32\\cmd.exe', 'NUMBER_OF_PROCESSORS': '2', 'PATHEXT': '.COM;.EXE;.BAT;.CMD;.VB...nce/test_session_management.py::test___multi_session___measure___reserves_multiple_sessions[configuration10] (setup)'})
key = 'ProgramW6432'
    def __getitem__(self, key):
        try:
            value = self._data[self.encodekey(key)]
        except KeyError:
            # raise KeyError with the original key value
```

Reply:
Anyway, I logged into the runner and ran `dir env:`, and this variable was set, as expected. You should not have to set it.

Reply:
Yes, I saw this error in pipeline that's why I overrided the key `ProgramW6432`. Now I have removed.

Reply:
Seems like tox environment is not able to access system environment variables, so I have passed in `ProgramW6432` to tox environment. This fixed the above key error issue.
`passenv = RUNNER_NAME, ProgramW6432`

Reply:
Thanks, that explains it. You may also need to pass `ProgramData`.

I posted a PR to remove the dependency on these environment variables: https://github.com/ni/measurementlink-python/pull/483

Reply:
I added ProgramData to the list.